### PR TITLE
Consistent format for package record

### LINKED ISSUE #881: manifest schema is different if using packrat
If I enable packrat:
```
options(rsconnect.packrat = TRUE)
```

And generate a manifest for an R Markdown doc:
```
rsconnect::writeManifest(appFiles=c("mydoc.Rmd"))
```

The manifest it generates has a different schema than if I do not enable packrat. In particular, the `description` of each package is an array rather than an object.

### With packrat:
```
  "packages": {
    "R6": {
      "Package": "R6",
      "Version": "2.5.1",
      "Source": "CRAN",
      "Repository": "http://rspm/default/__linux__/focal/latest",
      "description": ["Package: R6", ... etc ]
```

### Without packrate:
```
  "packages": {
    "R6": {
      "Package": "R6",
      "Version": "2.5.1",
      "Source": "CRAN",
      "Repository": "http://rspm/default/latest",
      "description": {
        "Package": "R6",
```

### PR DESCRIPTION
**Pull request by hadley**
Fixes #881

### CODE REVIEW

File: R/bundlePackageRenv.R [hunk 1]
```diff
@@ -5,8 +5,8 @@ snapshotRenvDependencies <- function(bundleDir,
 
   old <- options(
     renv.verbose = FALSE,
-    renv.consent = TRUE,
-    pkgType = "source")
```
Comment (hadley) [inline R/bundlePackageRenv.R:9]: Not sure this is correct; need to discuss.

Comment (hadley) [reply R/bundlePackageRenv.R:9]: Talked this over with @kevinushey, and I think the new behaviour is more correct — this should reduce snapshot time in the worst case where you have an existing lockfile and your library is out of sync.

File: R/bundlePackage.R [hunk 2]
```diff
@@ -93,3 +93,7 @@ availablePackages <- function(repos) {
     filters = getOption("available_packages_filters", default = "duplicates")
   )
 }
+
+package_record <- function(name, lib_dir = NULL) {
```
Comment (aronatkins) [inline R/bundlePackage.R:97]: Is it worth a test confirming the shape of the resulting JSON?

**Review by aronatkins — APPROVED** [review]

**Comment by aronatkins** [comment]
One other question: Do we want to remove the additional `Package=R6` and `Version=2.5.1` fields? They are not produced by the CRAN release.

**Comment by hadley** [comment]
@aronatkins I have a vague recollection we added those so that we could eventually drop the `description` entries altogether.

**Comment by aronatkins** [comment]
> I have a vague recollection we added those so that we could eventually drop the `description` entries altogether.

Hm. I don't recall. Maybe that was when we were thinking of always including the `renv.lock` and using that to drive the restore? Today, the `description` is needed to recalculate cache location since it captures dependencies.

**Comment by hadley** [comment]
Ok, if it doesn't help, then we might as well drop them.

**Review by aronatkins — APPROVED** [review]

**Pull request merged**